### PR TITLE
✨ Add button group component to unified components react [CB-15720]

### PR DESCRIPTION
**Pull request by cedricengagor**
Requires https://github.com/ClarabridgeInc/unified-ui/pull/87 to be merged and
released of course.

Fixes [CB-15720](https://clarabridge.atlassian.net/browse/CB-15720)

### CODE REVIEW

**Comment by cedricengagor** [comment]
Checks are going to fail until the linked PR is merged

File: src/Buttons/ButtonGroup.tsx [hunk 1]
```diff
@@ -6,7 +7,7 @@ export type ButtonGroupProps = {
 
 export default function ButtonGroup (props: PropsWithChildren<ButtonGroupProps>): ReactElement {
     return (
-        <div className={`flex ${props.className || ''}`}>
+        <div className={`btn-group ${props.className || ''}`}>
```
Comment (BenSeDev) [inline src/Buttons/ButtonGroup.tsx:10]: Q: Why no utility classes?

Comment (cedricengagor) [reply src/Buttons/ButtonGroup.tsx:10]: A button group is a css class that directly affects it's passed children, doing this with utilities is impossible. But hey, if you look at the other PR you'll see that I used utilities to generate this new CSS 🙇

**Review by thaoms — APPROVED** [review]

**Review by matthias-deschoenmacker — APPROVED** [review]

File: package.json [hunk 2]
```diff
@@ -32,7 +32,7 @@
     "homepage": "https://github.com/engagor/unified-components-react#readme",
     "peerDependencies": {
         "@clarabridge/unified-icons": "^8.11.1",
-        "@clarabridge/unified-ui": "6.0.2-alpha.42",
+        "@clarabridge/unified-ui": "ClarabridgeInc/unified-ui#improvement-cb-15720-button-groups-never-merge",
```
Comment (cedricengagor) [inline package.json:35]: REQ: Change to correct version once testing is complete

**Comment by cedricengagor** [comment]
@marcusrelacion / @goOve here is the test link you can use 🙇 => https://60df3699887198003bdfdd0e-vctuxvdjae.chromatic.com/

**Review by BenSeDev — APPROVED** [review]

**Review by goOve — APPROVED** [review]

**Pull request merged**